### ISSUE TITLE
How Do I get LinkedIn USER_TOKEN and USER_SECRET

### ISSUE THREAD

**Issue by dbl001** [ptwobrussell/Mining-the-Social-Web-2nd-Edition#274]
USER_TOKEN = ''
USER_SECRET = ''

These fields are not displayed in my LinkedIn Developer screen.

I only see:

Authentication Keys

    Client ID:
    	xxx
    Client Secret:
    	yyy

**Comment by ptwobrussell** [comment]
Try the two fields you mention. It may be the case that the user interface has just changed but the underlying intention is the same

---Sent from Boxer | http://getboxer.com

On May 21, 2015 at 1:38:45 PM CDT, dbl001 <notifications@github.com> wrote:USER_TOKEN = '' USER_SECRET = ''  These fields are not displayed in my LinkedIn Developer screen.  I only see:  Authentication Keys  Client ID: xxx Client Secret: yyy    —Reply to this email directly or view it on GitHub.

**Comment by dbl001** [comment]
auth = linkedin.LinkedInDeveloperAuthentication(CONSUMER_KEY, CONSUMER_SECRET,
                                USER_TOKEN, USER_SECRET,
                                RETURN_URL,
                                permissions=linkedin.PERMISSIONS.enums.values())

So try using:
CONSUMER_KEY, CONSUMER_SECRET (which is available from LinkedIn)
in the USER_TOKEN, USER_SECRET fields of the linkedin.LinkedInDeveloperAuthentication method?

**Comment by dbl001** [comment]
It didn't work.  Any other ideas?

**Comment by dbl001** [comment]
I got this document from LinkedIn:

https://developer.linkedin.com/blog/posts/2015/transition-faq

Does it mean your LinkedIn code will no longer work?

**Comment by bashyroger** [comment]
Yes, The Linkedin Api is (in my opinion) basically closed now, except when:
1) You are a partner / bring a lot of money
2) You use the api to complement your company's career pages, though you first have to apply for that....

**Comment by dbl001** [comment]
Do you think I’m allowed to screen scrape the data?

**Comment by bashyroger** [comment]
Officially, no: 
https://help.linkedin.com/app/answers/detail/a_id/56347/~/linkedins-scraping-software-policy
Technically, it is possible, for instance using http://scrapy.org/
If you do this you have to be very careful not to issue too many request, otherwise you will be blocked for a certain amount of time. And you even risk having your account shut down if you are identifiable...

**Comment by dbl001** [comment]
Is this the correct iPython notebook interface?  I can’t seem to find the notebooks.
I’m running the notebook server natively on my Mac with an Anaconda stack.
conda update ipython ipython-notebook ipython-qtconsole

**Comment by dbl001** [comment]
Never mind!  I started $ ipython notebook in the wrong directory.

**Comment by dbl001** [comment]
Is this LinkedIn people-search API Query also closed?

u = "https://api.linkedin.com/v1/people-search/%s:(relation-to-viewer:(related-connections))?format=json" % result["id"]

**Comment by ptwobrussell** [comment]
As I understand it, the LinkedIn API is effectively closed for data mining
as outlined in Chapter 3 of MTSW2E. Most of the examples still work,
however, given that they leverage the CSV download of your contacts. As
someone else has pointed out, you can page scrape, but that's violating the
ToS, so do it at your own peril ;)

**Comment by dbl001** [comment]
Thanks.

I got the CSV downloads yesterday.

David-Laxers-MacBook-Pro:testing davidlaxer$ ls -l ~/Downloads/LinkedInDataExport_07-28-2015/
total 3296
-rwxr-xr-x@ 1 davidlaxer  staff   13673 Jul 28 14:35 -1169023372.jpg
-rwxr-xr-x@ 1 davidlaxer  staff    3447 Jul 28 14:35 1381476224.jpg
-rwxr-xr-x@ 1 davidlaxer  staff     359 Jul 28 14:35 1495406907.jpg
-rwxr-xr-x@ 1 davidlaxer  staff     823 Jul 28 14:35 1886570078.jpg
-rwxr-xr-x@ 1 davidlaxer  staff     122 Jul 28 14:35 Account Status History.csv
-rwxr-xr-x@ 1 davidlaxer  staff    4604 Jul 28 14:35 Ad Targeting.csv
-rwxr-xr-x@ 1 davidlaxer  staff    2776 Jul 28 14:35 Ads Clicked.csv
-rwxr-xr-x@ 1 davidlaxer  staff     144 Jul 28 14:35 Certifications.csv
-rwxr-xr-x@ 1 davidlaxer  staff    2334 Jul 28 14:35 Comments.csv
-rwxr-xr-x@ 1 davidlaxer  staff  183960 Jul 28 14:35 Connections.csv
-rwxr-xr-x@ 1 davidlaxer  staff     388 Jul 28 14:35 Courses.csv
-rwxr-xr-x@ 1 davidlaxer  staff     627 Jul 28 14:35 Education.csv
-rwxr-xr-x@ 1 davidlaxer  staff     406 Jul 28 14:35 Email Addresses.csv
-rwxr-xr-x@ 1 davidlaxer  staff   46592 Jul 28 14:35 Endorsement Received Info.csv
-rwxr-xr-x@ 1 davidlaxer  staff   22800 Jul 28 14:35 Likes.csv
-rwxr-xr-x@ 1 davidlaxer  staff   25154 Jul 28 14:35 Login Attempts.csv
-rwxr-xr-x@ 1 davidlaxer  staff     556 Jul 28 14:35 Mobile Applications.csv
-rwxr-xr-x@ 1 davidlaxer  staff    3674 Jul 28 14:35 Positions.csv
-rwxr-xr-x@ 1 davidlaxer  staff     812 Jul 28 14:35 Profile.csv
-rwxr-xr-x@ 1 davidlaxer  staff    7264 Jul 28 14:35 README.txt
-rwxr-xr-x@ 1 davidlaxer  staff    6016 Jul 28 14:35 Recommendations Received.csv
-rwxr-xr-x@ 1 davidlaxer  staff     326 Jul 28 14:35 Registration Info.csv
-rwxr-xr-x@ 1 davidlaxer  staff  623804 Jul 28 14:35 Search Queries.csv
-rwxr-xr-x@ 1 davidlaxer  staff     446 Jul 28 14:35 Security Challenges.csv
-rwxr-xr-x@ 1 davidlaxer  staff  666770 Jul 28 14:35 Shares.csv
-rwxr-xr-x@ 1 davidlaxer  staff    1234 Jul 28 14:35 Skills.csv
David-Laxers-MacBook-Pro:testing davidlaxer$ 

Do you know if it includes: (relation-to-viewer:(related-connections))?


also, Is there an update to MTSW2E?
O’Reilly shows the last revision was July 10, 2014.

**Comment by Sherleen** [comment]
I have the same problem, so I want to know how to solve this problem.Can you tell me? Thank you so much~
